config/src/config/node_config_loader.rs

```diff
@@ -126,6 +126,18 @@ fn optimize_and_sanitize_node_config(
     NodeConfig::sanitize(node_config, node_type, chain_id)
 }
 
+pub fn sanitize_node_config(node_config: &mut NodeConfig) -> Result<(), Error> {
+    let node_type = NodeType::extract_from_config(node_config);
```

Comment:
nit: it might be worth adding a helper function to do the node type and chain ID extraction so that we don't have the duplication with the function above.